src/extension/android/androidPlatform.ts

```diff
@@ -84,24 +75,48 @@ export class AndroidPlatform extends GeneralMobilePlatform {
             runOptions.nodeModulesRoot,
             this.logger,
         );
-        this.emulatorManager = new AndroidEmulatorManager(this.adbHelper);
+        this.targetManager = new AndroidTargetManager(this.adbHelper);
     }
 
-    // TODO: remove this method when sinon will be updated to upper version. Now it is used for tests only.
-    public setAdbHelper(adbHelper: AdbHelper): void {
-        this.adbHelper = adbHelper;
+    public showDevMenu(deviceId?: string): Promise<void> {
+        return this.adbHelper.showDevMenu(deviceId);
+    }
+
+    public reloadApp(deviceId?: string): Promise<void> {
+        return this.adbHelper.reloadApp(deviceId);
     }
 
-    public async resolveVirtualDevice(target: string): Promise<IAndroidEmulator | null> {
-        if (!target.includes("device")) {
-            const emulator = await this.emulatorManager.startEmulator(target);
-            if (emulator) {
-                GeneralMobilePlatform.setRunArgument(this.runArguments, "--deviceId", emulator.id);
+    public async getTarget(): Promise<AndroidTarget> {
+        if (!this.target) {
+            const onlineTargets = await this.adbHelper.getOnlineTargets();
+            const onlineTargetsBySpecifiedType = onlineTargets.filter(target => {
+                switch (this.runOptions.target) {
+                    case TargetType.Simulator:
+                        return target.isVirtualTarget === true;
+                    case TargetType.Device:
+                        return target.isVirtualTarget === false;
+                    default:
+                        return true;
+                }
```

Comment:
May we simplify here like this?
```suggestion
                return this.runOptions.target !== TargetType.Device
```